src/react-components/room/useRoomLoadingState.js

```diff
@@ -53,6 +69,7 @@ const messages = defineMessages({
 export function useRoomLoadingState(sceneEl) {
   // Holds the id of the current
   const loadingTimeoutRef = useRef();
+  const lazyLoadMedia = useRef(!!APP.store.state.preferences.lazyLoadSceneMedia);
```

Comment:
I don't think using `useRef()` here is buying you anything, the only thing you are passing it to is the `useReducer()` initial state, but you aren't actually passing the reference to it, just the current value (it would be pretty weird to pass a mutable reference to a reducer anyway)

I assume what you were aiming to do with this was to have this re-render if that pref changed, but as written this wont actually do that. The "correct" way to handle this would be listening for `statechanged` event in a `useEffect()` like we do for example here https://github.com/mozilla/hubs/blob/master/src/react-components/room/useResizeViewport.js#L44-L64 and then dispatching an action to update the reducer state. I don't think its worth doing this since you only see this screen on first load anyway, so there is no real way a user is going to be able to change the preference. Even if we did show it multiple times, the user would never change the preferences during a given load. I would just get rid of `useRef()` and directly access the value (be sure to update all the locations accessing `.current`.

Reply:
Thanks for that, updated.